networkx/algorithms/tests/test_hierarchy.py

```diff
@@ -39,7 +39,8 @@ def test_hierarchy_weight():
     assert nx.flow_hierarchy(G, weight="weight") == 0.75
 
 
-def test_hierarchy_empty_graph():
-    G = nx.DiGraph()
-    G.add_node(1)
-    pytest.raises(nx.NetworkXError, nx.flow_hierarchy, G)
+@pytest.mark.parametrize("n", (0, 1, 3))
```

Comment:
Although these graphs have the same property in that they contain no edges, there's still enough distinction between these cases that it doesn't hurt to test them! In other words, I think that each one of these cases is "special" enough to be worth testing. For example, there are plenty of places in NX where there are checks on the number of nodes; with the numbers zero (null graph) and one (a singleton node) being special.

In the future, someone might decide to completely re-implement `flow_hierarchy` - during their rewriting, they may replace the `is_empty` check with something like `G.number_of_nodes() < 1` (or `<=`). In those cases, the test suite would've passed for some, but missed the case where there are multiple nodes with no edges. One of the primary uses of the test suite is to ensure that changes to code don't unintentionally modify behavior - so tests that cover potential behavior changes are valuable!

All in all the above hypothetical is not a very likely scenario, but IMO it's a principle of good test design to think about the semantics you are trying to enforce, and make sure you probe them as completely as possible. Of course, there's always a tradeoff with test runtime, but in this case adding an extra check is extremely cheap so that's not a concern.